src/main/convert_expressions.c

```diff
@@ -222,17 +209,18 @@ static as_status get_expr_size(int *size_to_alloc, int *intermediate_exprs_size,
         [VAL] = EXP_SZ(as_exp_val(
             NULL)), // NOTE if I don't count vals I don't need to subtract from other ops // MUST count these for expressions with var args.
         [EQ] = EXP_SZ(as_exp_cmp_eq(
-            {}, {})), // ^ TODO implement a less wastefull solution.
-        [NE] = EXP_SZ(as_exp_cmp_ne({}, {})),
-        [GT] = EXP_SZ(as_exp_cmp_gt({}, {})),
-        [GE] = EXP_SZ(as_exp_cmp_ge({}, {})),
-        [LT] = EXP_SZ(as_exp_cmp_lt({}, {})),
-        [LE] = EXP_SZ(as_exp_cmp_le({}, {})),
-        [CMP_REGEX] = EXP_SZ(as_exp_cmp_regex(0, "", {})),
-        [CMP_GEO] = EXP_SZ(as_exp_cmp_geo({}, {})),
-        [AND] = EXP_SZ(as_exp_and({})),
-        [OR] = EXP_SZ(as_exp_or({})),
-        [NOT] = EXP_SZ(as_exp_not({})),
+            as_exp_nil(),
+            as_exp_nil())), // ^ TODO implement a less wastefull solution.
+        [NE] = EXP_SZ(as_exp_cmp_ne(as_exp_nil(), as_exp_nil())),
```

Comment:
All the as_exp_nil() calls reduce readability in an already dense file. I think a macro that wraps `as_exp_nil` would help. Maybe `#define NIL as_exp_nil()`.

Reply:
done